packages/core/src/components/drawer/_drawer.scss

```diff
@@ -0,0 +1,128 @@
+// Copyright 2018 Palantir Technologies, Inc. All rights reserved.
+// Licensed under the terms of the LICENSE file distributed with this project.
+
+@import "~@blueprintjs/icons/src/icons";
+@import "../../common/mixins";
+@import "../../common/react-transition";
+@import "../../common/variables";
+
+$drawer-margin: ($pt-grid-size * 3) 0 !default;
+$drawer-padding: $pt-grid-size * 2 !default;
+
+.#{$ns}-drawer {
+  @include react-transition-phase(
+    "#{$ns}-overlay",
+    "enter",
+    (transform: (translateX(100%), translateX(0))),
+    $pt-transition-duration * 2,
+    $pt-transition-ease,
+    $before: "&"
+  );
+  @include react-transition-phase(
+    "#{$ns}-overlay",
+    "exit",
+    (transform: (translateX(100%), translateX(0))),
+    $pt-transition-duration,
+    $before: "&"
+  );
+  display: flex;
+  flex-direction: column;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  margin: 0;
+  box-shadow: $pt-elevation-shadow-4;
+  background: $white;
+  width: 50vw;
+  padding: 0;
+
+  &:focus {
+    outline: 0;
+  }
+
+  &.#{$ns}-vertical {
+    @include react-transition-phase(
+      "#{$ns}-overlay",
+      "enter",
+      (transform: (translateY(100%), translateY(0))),
+      $pt-transition-duration * 2,
+      $pt-transition-ease,
+      $before: "&"
+    );
+    @include react-transition-phase(
+      "#{$ns}-overlay",
+      "exit",
+      (transform: (translateY(100%), translateY(0))),
+      $pt-transition-duration,
+      $before: "&"
+    );
+    top: auto;
+    bottom: 0;
+    left: 0;
+    width: 100vw;
+    height: 50vh;
```

Comment:
i was wrong about viewport units, this actually works much better with %s. disable the "Use `Portal`" switch and you'll see what i mean. it's way too big with vh/w but works perfectly when changed to %.